File: python/680.py
class Solution:
    def validPalindrome(self, s):
        """
        :type s: str
        :rtype: bool
        """
        start = 0
        end = len(s) - 1
        flag = True
        def isValid(start,end,s,flag):
            while start < end:
                if s[start] == s[end]:
                    start += 1
                    end -= 1
                else:
                    if flag:
                        return isValid(start+1,end,s,False) or isValid(start,end-1,s,False)
                    else:
                        return False
            return True

        return isValid(start,end,s,flag)

File: java/680.java
class Solution {
    public boolean validPalindrome(String s) {
        int l = s.length();
        for(int i = 0;i < l/2;i++){
            if(s.charAt(i) != s.charAt(l-i-1)){
                int j = l-i-1;
                return helper(s,i,j-1) || helper(s,i+1,j);
            }
        }
        return true;
        
    }
    
    public boolean helper(String s, int i,int j){
        for(int k = i; k <= i + (j-i)/2;k++){
            if(s.charAt(k) != s.charAt(j-k+i)){
                return false;
            }
        }
        return true;
    }
}